motor/discovery/azure/mql_asset_objects.go

```diff
@@ -92,7 +96,19 @@ func computeInstances(m *MqlDiscovery, subscription string, tc *providers.Config
 					service:      "compute",
 					objectType:   "vm",
 				},
-			}, tc))
+			}, tc)
+		for _, ip := range vm.PublicIpAddresses {
```

Comment:
i think if you add the ip connections here then you'll run into weirdness later when we do resources as assets.

because one is the inside (os) scan - the ones we need the ip for, and the other side of this asset is the outside scan (configuration, from the api)

Reply:
i may be wrong, i'm staring at the code still. haven't tested it myself

Reply:
ok, sorry, new day brain, i can explain now:

- when we have inside/outside scan available for an asset, we want to leave the inside and outside scans as two different assets (we'll do some work with asset relationships to show them as one in console)
- so one asset does the configuration (outside scan)
- other asset does the os (inside) scan

so you would have two discovery targets for the compute instances, "instances" and "instances-api" (or something like that)

that said, we can also just change that later on 🤷

Reply:
code looks good! i'm gonna hit approve on this, we can always move the connections logic around later